pocs/scheduler/dispatch.py

```diff
@@ -95,14 +95,20 @@ def get_observation(self, time=None, show_all=False, reread_fields_file=False):
             self.current_observation = self.observations[top_obs[0]]
             self.current_observation.merit = top_obs[1]
         else:
+            # Favor the current observation if still available
             if self.current_observation is not None:
-                # Favor the current observation if still available
+                # Check that the target field is still avialable
                 end_of_next_set = time + self.current_observation.set_duration
-                if end_of_next_set < common_properties['end_of_night'] and \
-                        self.observation_available(self.current_observation, end_of_next_set):
-
-                    self.logger.debug("Reusing {}".format(self.current_observation))
-                    best_obs = [(self.current_observation.name, self.current_observation.merit)]
+                can_complete_set = end_of_next_set < common_properties['end_of_night']
+                still_up = self.observation_available(self.current_observation, end_of_next_set)
```

Comment:
Hmm, you're right that there is a problem. If the observation is vetoed because it is not available for 60 exposures it will stay vetoed even if it could do 10 exposures (the set size).

We added this in a rush to support the sequential mode. I'll give it a little more though. Marking as WIP. Thanks!